### ISSUE TITLE
Get product review list added by customer - Using Bagisto Store Front Rest API

### ISSUE THREAD

**Issue by rohit-ghoghari** [bagisto/rest-api#261]
### Is your feature request related to a problem? Please describe.

We have REST API for adding reviews in products but when I check we don't have REST API to get a list of reviewers list.

### Describe the solution you'd like

We need to create one GET REST API to receive all product review lists.

**Comment by amit-webkul** [comment]
I have checked and found that this API is already in our swagger with the name CustomerReviews. I have added a screenshot for a reference:


![product-review-shop](https://github.com/bagisto/rest-api/assets/121479029/e518f8f6-98dc-42bf-8334-25317e61cf36)

**Comment by rohit-ghoghari** [comment]
Dear Amit, Thanks for your response.

I need reviews of by products not by customers. In headless development, I need all reviews of selected product by its Id. Right now API is providing Reviews of selected customer.

**Comment by amit-webkul** [comment]
Hello @rohit-ghoghari,

I hope this message finds you well.

Regarding your request to view all reviews by product ID, I have investigated and found that this feature is currently unavailable. At present, we can only store reviews by product ID. I have attached an image to illustrate this [Link](https://prnt.sc/fzFhut6aRI6c).

To address this issue, I will create a new pull request to implement the functionality for retrieving all reviews by product ID. I will share the details with you once it is completed.

Thank you for bringing this to our attention.

Best regards,

**Comment by rohit-ghoghari** [comment]
Okay Amit,

I can help you with this if you will allow me. I will create new pull request for this enhancement and you only need to merge it.

Thanks

**Comment by amit-webkul** [comment]
> Okay Amit,
> 
> I can help you with this if you will allow me. I will create new pull request for this enhancement and you only need to merge it.
> 
> Thanks

Yes you can proceed with PR. Just make sure to follow all the `coding standard` and `code quality`.

We will be glad if you can contribute to Bagitso rest API.

Thanks

**Comment by rohit-ghoghari** [comment]
Hello Amit,

I have sent a pull request for product review REST API enhancement.

[Request link](https://github.com/bagisto/rest-api/pull/331) 

Please review.

Thanks

**Comment by nishant-webkul** [comment]
@amit-webkul we need to create the REST API to get all product review list added by the customer. 

Checked in both v2.2.1 and v2.2.2

**Reference Screenshot:**

![Bagisto-Rest-Shop-API-Documentation-12-30-2024_01_10_PM](https://github.com/user-attachments/assets/24f845b2-8b36-4890-83dd-d68ed0ec2046)
